src/Service.Tests/SqlTests/GraphQLMutationTests/MultipleMutationIntegrationTests.cs

```diff
@@ -0,0 +1,327 @@
+                    message: "Missing value for required column: publisher_id for entity: Book at level: 1.",
+                    statusCode: DataApiBuilderException.SubStatusCodes.BadRequest.ToString()
+                );
+        }
+
+        /// <summary>
+        /// Test to validate that for 'create one' mutations, we throw an appropriate exception when the user request provides
+        /// multiple sources of value for a referencing column. There are 3 sources of values for a referencing column:
+        /// -> Explicit value assignment to the referencing column,
+        /// -> Value derived from the parent referenced entity,
+        /// -> Providing value for inserting record in the referenced target entity, in which case value for referencing column
+        /// is derived from the value of corresponding referenced column after insertion in the referenced entity.
+        /// When the user provides more than one of the above, we throw an exception.
+        /// </summary>
+        [TestMethod]
+        public async Task PresenceOfMultipleSourcesOfTruthForReferencingColumnForCreateOneMutations()
+        {
+            // For a relationship between Book (Referencing) - Publisher (Referenced) defined as books.publisher_id -> publisher.id,
+            // consider the request input for Book:
+            // 1. An explicit value is provided for referencing field publisher_id.
+            // 2. A value is assigned for the target (referenced entity) via relationship field ('publisher') which will
+            // give back another value for publisher_id.
+            string createOneBookMutationName = "createbook";
+            string multipleCreateOneBookWithPublisher = @"mutation {
+                    createbook(item: { title: ""My New Book"", publisher_id: 1234, publishers: { name: ""New publisher""}}) {
+                        id
+                        title
+                    }
+                }";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(
+                query: multipleCreateOneBookWithPublisher,
+                queryName: createOneBookMutationName,
+                isAuthenticated: true,
+                variables: null,
+                clientRoleHeader: "authenticated");
+
+            // Validate that an appropriate exception is thrown because we have two possibly conflicting sources of
+            // values for the referencing field 'publisher_id'.
+            SqlTestHelper.TestForErrorInGraphQLResponse(
+                    actual.ToString(),
+                    message: "Found conflicting sources of values for the field: publisher_id for entity: Book at level: 1." +
+                    "Source 1: entity: Book ,Source 2: Relationship: publishers.",
+                    statusCode: DataApiBuilderException.SubStatusCodes.BadRequest.ToString()
+                );
+
+            // For a relationship between Review (Referencing) - Book (Referenced) defined as reviews.book_id -> books.id,
```

Comment:
When reading the mutation left to right, it would be helpful if this description were consistent with reading the mutation:
 For a relationship between Book (Referenced) - Review (Referencing)  defined as books.id <- reviews.book_id